mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -643,14 +643,25 @@ impl mayastor_server::Mayastor for MayastorSvc {
         trace!("{:?}", args);
 
         let rx = rpc_submit::<_, _, nexus_bdev::Error>(async move {
+            let mut nexus_list: Vec<NexusV2> = Vec::new();
+
+            for n in instances() {
+                if n.state.lock().deref() != &nexus_bdev::NexusState::Init {
+                    let mut o = n.to_grpc_v2();
+
+                    // Get ANA state only for published nexuses.
+                    if let Some(Protocol::Nvmf) = n.shared() {
```

Comment:
Could we not do this on to_grpc_v2?

Reply:
Why

Reply:
Yes, so it's contained, and if something else were to call `to_grpc_v2()` it would not have to also do this extra set of setting up the ana state?

Reply:
No problem with that, it only requires making the conversion routine async, as ANA state retrieval is async.

Reply:
Done.